### ISSUE TITLE
[feature request] parsed object or parse Reviver function contains line and column ranges for keys & values.

### ISSUE THREAD

**Issue by AlexLeung** [kaelzhang/node-comment-json#12]
**Why this feature is better to be included in comment-json? Please describe the scenario**

If I'm writing a tool which parses a JSON file, then validates the file, the presence of line and column number info can be used to display user-friendly error messages and eventually could be used to create an IDE/editor plugin.

**Provide 1-2 code examples that the usage of the new feature:**

```ts
import { parse } from 'comment-json';
const sourceFile = "some.json";
const someJsonString = require('fs').readFileSync(sourceFIle).toString();
parse(someJsonString, (key: number, value: any, keyRange: SourceRange, valueRange: SourceRange) => {
    if (valueIsInvalid(value)) {
        // Below is pseudo code I don't know VSCode plugin APIs well.
        vsCodePlugin.drawRedErrorLine(sourceFile, keyRange);
    }
})
```

definitions for comment-json gets this addition

```ts
interface SourceRange {
    startLine: number;
    startColumn: number;
    endLine: number;
    endColumn: number;
}
```

**Are you willing to submit a pull request to implement this rule?**

yes eventually.

**Comment by kaelzhang** [comment]
The error object thrown by `comment-json` parser contains the line and column where the error occurs, which is not well documented though.

```js
import {parse} from 'comment-json'

try {
   parse(someInvalidJsonString)
} catch (e) {
   console.log(e instanceof SyntaxError)  // true
   console.log(e.line) // this will print the line of the unexpected token
   console.log(e. column) // this will print the column of the unexpected token
}
```

**Comment by AlexLeung** [comment]
I was using this feature already actually. I was hoping that the line and column numbers could be returned even when there is no error.

**Comment by kaelzhang** [comment]
Landed in `2.3.0`

The `CommentToken` will now includes a `loc` property
